### ISSUE TITLE
How to use KPConv as a feature extractor?

### ISSUE THREAD

**Issue by 1zb** [nicolas-chaulet/torch-points3d#174]
Hello Team,

Thanks for your wonderful work. I have work that needs per-point feature and I decided to use KPConv as our backbone. With your framework, it is easy to write a segmentation network, due to the encapsulation. But I want to only use the features before MLP is applied. How can I achieve this?

Thanks!

**Comment by tchaton** [comment]
Dear @1zb,

Thanks for contacting us. Which task are you interested in ?

When you write your model logic within models/{task}/{your_model}, you can access everything there easily and modify the network for your own usage.
However, UNetBase constructor does several things in the backend to simplify your life and if you plan to have a different overall architectures, you should have a look to it.

Finally, if you plan to use the framework, send me an email and we will add you to our Slack ;)

Best,
T.C

**Comment by 1zb** [comment]
Thank you! It's really helpful.

I want to use the per-point feature for per-point embedding and then a customized loss function is applied. To take an example, for `torchvision.models.AlexNet`, I will only need `alexnet.features`.

This is my email address. biao.zhang@kaust.edu.sa

Best,
Biao

**Comment by tchaton** [comment]
Hey @1zb,

The framework is still not a library, but it can be easily done. We plan to open sourced pre-trained models on the already implemented datasets, for people to use.
I also added you on Slack.

Best,
T.C

**Comment by dragonlong** [comment]
Hi @tchaton ,

Thanks for this amazing integration, could you also add me to slack? I am also interested in this feature. 

Best,
Xiaolong

**Comment by sayakgis** [comment]
> Dear @1zb,
> 
> Thanks for contacting us. Which task are you interested in ?
> 
> When you write your model logic within models/{task}/{your_model}, you can access everything there easily and modify the network for your own usage.
> However, UNetBase constructor does several things in the backend to simplify your life and if you plan to have a different overall architectures, you should have a look to it.
> 
> Finally, if you plan to use the framework, send me an email and we will add you to our Slack ;)
> 
> Best,
> T.C

Is it possible to add me in slack, email: sayakgis@gmail.com, I plan to use the library for aerial lidar point cloud segmentation.

**Comment by AlphaPav** [comment]
Dear @tchaton ,
Could you also add me in slack, email:chulinxie@gmail.com?
I plan to use KPConv for feature extraction. Thank you!

**Comment by adithyamurali** [comment]
Hi Thomas, thanks so much for the amazing library!

I was reading through the KPConv example: https://github.com/nicolas-chaulet/torch-points3d/blob/master/notebooks/PartSegmentationKPConv.ipynb and had some questions:

Is this example using pre-trained weights? If so where are the weights being loaded from? Are they trained from the ShapeNet Part segmentation task? I'm planning to use `torch_points3d.applications.pointnet2.PointNet2` and I'm assuming it should follow a similar structure as KPConv.

Do you mind adding me to the slack group as well? My email is amurali@andrew.cmu.edu

Thanks!

**Comment by tchaton** [comment]
Dear @adithyamurali,

Thanks for having a look at our notebook !

This example is not a pre-trained example. 
However, we have a built-in ModelCheckpoint if you use train.py script as within the examples.

PointNet2 differs slightly from KPConv.
KPConv implements a partial_dense format.
PointNet2, RSConv implements a dense format.

Here is an explanation about the format: https://torch-points3d.readthedocs.io/en/latest/src/advanced.html#data-formats-for-point-cloud

I just added you to our Slack !

Best regards,
T.C
